.github/workflows/build_and_test.yml

```
@@ -1092,12 +1093,46 @@ jobs:
               }
 
               console.log(`Release for tag ${tag} does not exist. Creating it...`);
+
+              // Get the merge commit message to extract the PR number
+              const commit = await github.rest.git.getCommit({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                commit_sha: context.sha,
+              });
+              const prMatch = commit.data.message.match(/Merge pull request #(\d+)/);
+              const prNumber = prMatch ? prMatch[1] : null;
+
+              // Find the previous release on the branch
+              const releases = await github.rest.repos.listReleases({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                per_page: 100,
+              });
+
+              const branchReleases = releases.data
+                .filter((release) => {
```

As you need only the first element you can use `find` here:

```suggestion
                .find((release) => {
```

---

then we need to sort before filtering. Not sure whether sorting all and finding first or filtering first, sorting the subset and taking that is faster...

---

Ah, yeah, you're right